ddtrace/profiling/collector/_memalloc_heap.c

```diff
@@ -21,6 +27,16 @@ typedef struct
 
 static heap_tracker_t global_heap_tracker;
 
+static uint32_t
+heap_tracker_next_sample_size(uint32_t sample_size)
+{
+    /* Get a value between [0, 1[ */
+    double q = (double)rand() / ((double)RAND_MAX + 1);
+    /* Get a value between ]-inf, 0[, more likely close to 0 */
+    double log_val = log2(q);
+    return (uint32_t)(log_val * (-log(2) * (sample_size + 1)));
```

Comment:
LGTM. Just a pedantic note: I don't know if the compiler optimises `log(2)`, but I suspect that it doesn't, so perhaps we can define a constant or even replace `log(2)` with `0.3`

Reply:
I prefer to keep `log(2)` here and leave optimization to the compiler, if it can.